featurebyte/api/feature_list.py

```diff
@@ -677,13 +678,29 @@ def _pre_save_operations(self, conflict_resolution: ConflictResolution = "raise"
                     feat = Feature.get(name=feat_name)
                     self.feature_objects[feat_name] = feat
                 except RecordRetrievalException:
-                    feature_payloads.append(FeatureCreate(**feat.dict(by_alias=True)))
+                    features.append(feat)
             if conflict_resolution == "raise" and not feat.saved:
-                feature_payloads.append(FeatureCreate(**feat.dict(by_alias=True)))
+                features.append(feat)
```

Comment:
**main**
`feat.dict(by_alias)` is expensive as it will prune the feature graph from the global query graph. Each feature pruning will involve one operation structure extraction & reconstruction of the pruned feature graph. `BatchFeatureCreate.create` combines all individual pruned graphs into a single combined graph.

**current branch**
`GlobalQueryGraph().crop(target_node_names=[feat.node_name for feat in features])` directly crops the global graph based on the given list of target nodes. The cropped graph is a combined graph of list of feature graphs.